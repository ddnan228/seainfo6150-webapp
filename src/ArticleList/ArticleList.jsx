import React from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx"
import PropTypes from "prop-types"

const ArticleList = props => {
  return (
    <ul>
      {props.articles.map(article => (
        <li>
          <ArticleListItem article ={article} />
        </li>
      ))}
    </ul>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired
};
export default ArticleList;
