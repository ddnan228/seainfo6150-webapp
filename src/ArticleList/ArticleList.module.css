.articleList {
  background-color: #8d9194;
  padding: 15px;
  margin: 15px;
  list-style: none;
  border-radius: 15px;
  border: 2px #545759 dotted;
  box-shadow: red 0px 0px 15px;

  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  column-gap: 20px;
  row-gap: 10px;
}

.articleBlock {
  display: grid;
  grid-auto-flow: row;
}
