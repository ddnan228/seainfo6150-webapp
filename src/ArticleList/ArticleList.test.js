import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";

describe("ArticleList test", () => {

  it("renders correctly", () => {
    const test = [
      {
        title: "The Statue of Liberty's torch heads to new museum",
        author: "Kate Farley",
        displayDate: "November 22nd 2018, 7:12 am",
        shortText: "The Statue of Liberty's original torch is getting a new home",
        image: {
          _url: "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg",
        },
      },
      {
        title: "You can take a pie on a plane",
        author: "Cathryn Lindsey",
        displayDate: "November 21st 2018, 2:57 pm",
        shortText: "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?",
        image: {
          _url: "https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg",
        },
      }
    ]

    const { container } = render(<ArticleList articles={Object.values(test)} />);
    expect(container).toMatchSnapshot();
  });
});
