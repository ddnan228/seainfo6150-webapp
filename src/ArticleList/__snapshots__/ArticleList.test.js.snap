// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList test renders correctly 1`] = `
<div>
  <ul
    class="articleList"
  >
    <li
      class="articleBlock"
    >
      <div
        class="listItem"
      >
        <div>
          <img
            class="imageStyle"
            src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
          />
          <h1
            class="articleTitle"
          >
            <u>
              The Statue of Liberty's torch heads to new museum
            </u>
          </h1>
        </div>
        
        <button
          class="itemButton"
        >
          Show more
        </button>
      </div>
    </li>
    <li
      class="articleBlock"
    >
      <div
        class="listItem"
      >
        <div>
          <img
            class="imageStyle"
            src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
          />
          <h1
            class="articleTitle"
          >
            <u>
              You can take a pie on a plane
            </u>
          </h1>
        </div>
        
        <button
          class="itemButton"
        >
          Show more
        </button>
      </div>
    </li>
  </ul>
</div>
`;
